networklibrary/src/main/java/com/eis/smsnetwork/SMSNetworkManager.java

```diff
@@ -0,0 +1,96 @@
+package com.eis.smsnetwork;
+
+import com.eis.communication.network.FailReason;
+import com.eis.communication.network.Invitation;
+import com.eis.communication.network.NetworkManager;
+import com.eis.communication.network.listeners.GetResourceListener;
+import com.eis.communication.network.listeners.InviteListener;
+import com.eis.communication.network.listeners.RemoveResourceListener;
+import com.eis.communication.network.listeners.SetResourceListener;
+import com.eis.smslibrary.SMSPeer;
+
+/**
+ * The manager class of the network.
+ *
+ * @author Edoardo Raimondi
+ * @author Marco Cognolato
+ */
+public class SMSNetworkManager implements NetworkManager<String, String, SMSPeer, FailReason> {
+
+    private SMSNetSubscriberList netSubscribers = new SMSNetSubscriberList();
```

Comment:
Does this have to be `SMSNetSubscriberList `, couldn't it be a `NetSubscriberList<SMSPeer>`? So that it's independent of the type of class used.

Reply:
well, `SMSNetSubscriberList implements NetSubscriberList<SMSPeer>`, wouldn't that be the same?
Afterall **SMS** net manager would net **SMS** net sub list, right?

Reply:
What if I wanted to use my own `NetSubscriberList<SMSPeer>` that orders peers by telephone number and pass it as a parameter of the constructor of `SMSNetworkManager` because I want to keep everything else the same?

Reply:
my god, you want too much.
what if I add my list which just purposely doesn't store anything?

I get that you want this to be really exapandable etc, but... isn't this too much?

Reply:
It is a lot of generalization indeed but doesn't cost too much and makes real use of interfaces as real usable things instead of placeholders for methods to be implemented.

Reply:
Alright, but, in case you forgot, this is gonna be a singleton, so... `SMSNetworkManager.getInstance(yourList)`? `SMSNetworkManager.getInstance().setList(yourList)`?

Reply:
`setup(yourList, yourDictionary)`?

Reply:
Definitely `setNetSubscriberList()` and `setNetDictionary()`

Reply:
Is this really going to be a singleton? I couldn't remember the pros and cons

Reply:
It's going to be a singleton only early on, to have something working really soon. We'll switch to something else (object pool?) when we know it's definitely working